pkg/vsphere/compute/placement/ranked.go

```diff
@@ -42,25 +37,76 @@ func (r rankedHosts) Len() int           { return len(r) }
 func (r rankedHosts) Swap(i, j int)      { r[i], r[j] = r[j], r[i] }
 func (r rankedHosts) Less(i, j int) bool { return r[i].score > r[j].score }
 
+// WeightConfiguration holds user-provided weights for different host metrics. These weights are
+// used to determine a host ranking.
+type WeightConfiguration struct {
+	memUnconsumedWeight float64
+	memInactiveWeight   float64
+}
+
 // RankedHostPolicy uses data from a MetricsProvider to decide on which host to power-on a VM.
 type RankedHostPolicy struct {
 	source performance.MetricsProvider
+	config WeightConfiguration
 }
 
-// NewRankedHostPolicy returns a RandomHostPolicy instance using the supplied MetricsProvider.
+// NewRankedHostPolicy returns a RandomHostPolicy instance using the supplied MetricsProvider with
+// the default weighting configuration.
 func NewRankedHostPolicy(s performance.MetricsProvider) *RankedHostPolicy {
-	return &RankedHostPolicy{source: s}
+	return NewRankedHostPolicyWithConfig(s, WeightConfiguration{
+		memInactiveWeight:   memDefaultInactiveWeight,
+		memUnconsumedWeight: memDefaultUnconsumedWeight,
+	})
+}
+
+// NewRankedHostPolicyWithConfig returns a RandomHostPolicy instance using the supplied MetricsProvider and
+// WeightConfiguration.
+func NewRankedHostPolicyWithConfig(s performance.MetricsProvider, wc WeightConfiguration) *RankedHostPolicy {
+	return &RankedHostPolicy{
+		source: s,
+		config: wc,
+	}
 }
 
 // CheckHost returns true if the host has adequate capacity to power on the VM, false otherwise.
-func (r *RankedHostPolicy) CheckHost(op trace.Operation, vm *vm.VirtualMachine) bool {
+func (r *RankedHostPolicy) CheckHost(op trace.Operation, host *object.HostSystem) bool {
```

Comment:
This doesn't match the current interface definition - if this PR compiles it means that you are not using the interface anywhere in the testing.

Reply:
Fixed, although we are still not testing with the actual interface yet.